src/libraries/System.Net.Quic/src/System/Net/Quic/QuicStream.cs

```diff
@@ -531,7 +531,6 @@ private unsafe int HandleEventPeerSendShutdown()
     }
     private unsafe int HandleEventPeerSendAborted(ref PEER_SEND_ABORTED data)
     {
-        _receiveBuffers.SetFinal();
```

Comment:
Why this should not be set? I look at the ReceiveBuffers but it is not quite clear to me what this does.

Reply:
It successfully completes the buffer, propagation of FIN flag. You want to set it only in successful scenarios.

So what happened here is that we would set it here, `ReadAsync` would see this, i.e. that it's completed successfully and finish successfully the reading side of the stream, before the following exception could get propagated.